nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -720,6 +727,77 @@ def screenshot(self, filename: str) -> None:
             if ret.returncode != 0:
                 raise Exception("Cannot convert screenshot")
 
+    @contextmanager
+    def record_audio(
+        self,
+        filename: str,
+        allow_silence: bool = False,
+        device: str = "audio_device",
+        sample_rate: int = 44100,
+        bit_depth: int = 16,
+        channels: int = 2,
+    ) -> Iterator[None]:
+        filename = add_extension_if_simple_name(filename, "wav")
+        out_path = self.out_dir / filename
+
+        info_capture_regex = re.compile(
+            r"^\[(\d+)\]: Capturing audio\(\d+,\d+,\d+\) to (.*): (\d+) bytes$",
+        )
+
+        def capture_info() -> Optional[Tuple[int, int]]:
+            output = self.send_monitor_command("info capture")
+            lines = output.split("\r\n")
```

Comment:
i suspect that `split()` does more what you want:

```python
>>> "\r\nbla\r\n\r\nblubb\r\n".split("\r\n")
['', 'bla', '', 'blubb', '']
>>> "\r\nbla\r\n\r\nblubb\r\n".split()
['bla', 'blubb']
```

Reply:
Should be `splitlines`, but yes.